src/Resources/contao/classes/SurveyQuestionConstantsum.php

```diff
@@ -203,7 +203,8 @@ protected function calculateCumulated(): void
 
             if (\is_array($arrAnswer)) {
                 foreach ($arrAnswer as $answerkey => $answervalue) {
-                    ++$cumulated[$answerkey][$answervalue];
+                    // ToDo: fix the following workaround
+                    @++$cumulated[$answerkey][$answervalue];
```

Comment:
Wouldn't it be better to do something like `$cumulated[$answerkey][$answervalue] = ($cumulated[$answerkey][$answervalue] ?? 0) + 1` than suppress the warning?

Reply:
yes sure - suppressing the warning is a bad solution. But for a quick solution I can't think of anything better. I think that under PHP8 every key of an array must be tested for existence, otherwise we get unpredictable warnings. I am not sure if your solution works. I believe that the $answerkey and $answervalue keys must always be tested for existence in the array.

Reply:
This is exactly what the null coalescing operator do :)

Reply:
I may not have been able to explain my concerns properly yet and in english 😊. Maybe I don't understand something right here... 
The null coalescing operator certainly checks for null, but that is - as it seems to me - **not our real problem here**. 
The many warnings do not arise **because a value** of an addressed key/cell in the array/matrix etc. is **undefined or null**, but because PHP 8 warns if the desired value in the array **cannot be addressed**. I think this is the basic problem. 
So for example, if I have an array $arr = [] and I run a test: empty($arr['key']), PHP always throws a warning if the key does not exist and therefore a construct like ++$arr[k1][k2][kn] will never work. So I have to test if the key is in the array before each access. This is how I have understood it so far. It gets more complicated with multidimensional structures like matrix or above... Actually you have to test each key for its existence and if false initialize the key separately. At least that's how I understood it so far... What do you think?

Reply:
something like that

    foreach ($arrAnswer as $answerkey => $answervalue) {
      if(array_key_exists($answerkey, $cumulated)) {
        if(array_key_exists($answervalue, $cumulated[$answerkey])) {
          // only in this particular case the inline increment operator should work as expected
          ++$cumulated[$answerkey][$answervalue];
        } else {
          // $answervalue does not exist, inizialize it
          $cumulated[$answerkey][$answervalue] = 0;
        }
      } else {
        // the 'base key' named $answerkey does not exists, inizialize is as an array
        $cumulated[$answerkey] = []; 
      }
    }